app2.py

```diff
@@ -0,0 +1,264 @@
+        return r
+
+    # 5
+    def _get_storage_info(self) -> json:
+        # Returns storage summary information regarding binaries, file store and repositories.
+        # Usage: GET /api/storageinfo
+        endpoint = f'storageinfo'
+        url = self._set_url(endpoint)
+        r = self.session.get(url, headers=self.session.headers)
+        return r.json()
+
+    # 6
+    def _create_repository(self) -> requests.status_codes:
+        # Notes: Requires Artifactory Pro
+        # Creates a new repository in Artifactory with the provided configuration. Supported by local, remote, virtual and federated repositories. 
+        # Usage: PUT /api/repositories/{repoKey}
+        repokey = input("Enter repository name: ")
+        if repokey == '' or repokey is None:
+            raise ValueError("repository name cannot be empty")
+
+        json = {
+            'rclass' : 'remote',
+            'url' : f'{self.url}',
+            'externalDependenciesEnabled': False # False (default, Applies to Docker repositories only)
+        }
+        endpoint = f'repositories/{repokey}'
+        url = self._set_url(endpoint)
+        r = self.session.put(url, headers=self.session.headers, json=json)
+        return r.status_code
+
+    # 7
+    # TODO:
+    def _update_repo(self):
+        pass
+
+    # 8
+    def _list_repositories(self) -> json:
+        # Returns a list of minimal repository details for all repositories of the specified type.
+        # Usage: GET /api/repositories[?type=repositoryType (local|remote|virtual|federated|distribution)]|
+        endpoint = 'repositories'
+        url = self._set_url(endpoint)
+        r = self.session.get(url, headers=self.session.headers)
+        return r.json()
+
+    # 9
+    def _change_group(self, group: str) -> str:
+        self.group = group
+
+    # 10
+    # TODO:
+    def _create_group(self):
+        pass
+
+    def _display_menu(self) -> None:
+        menu_options = {
+            1: "Ping",
+            2: "Artifactory Version",
+            3: "Create User",
+            4: "Delete User",
+            5: "Get Storage Info",
+            6: "Create Repository",
+            7: "Update Repository",
+            8: "List Repositories",
+            9: f"Change Group (current group: '{self.group}')",
+            10: "Create Group",
+            11: "Exit",
+            12: "Print Token"
+            }
+
+        for option in menu_options.keys():
+            if option < 10:     
+                print(option, " --", menu_options[option])
+            else:
+                print(option, "--", menu_options[option])
+
+    def menu(self) -> None:
+        while True:
+            print()
+            self._display_menu()
+
+            option = ''
+            try:
+                option = int(input("Enter your choice: "))
+            except:
+                print("Wrong input. Please enter a number ...\n")
+    
+            if option == 1:
```

Comment:
Instead of `if/else`, you can implement a dict with functions:
`{PING: self._system_ping, ARTIFACTORY_VERSION: self._system_version}...`

And then `option = op_switch_case.get(<user_input>, None)`
It will shorten your code significantly and can be more generic, for example:
the print after each op can be: `print(f"{option.value} {version}")`